src/librbd/image/AttachParentRequest.cc

```diff
@@ -51,7 +50,7 @@ void AttachParentRequest<I>::handle_attach_parent(int r) {
   auto cct = m_image_ctx.cct;
   ldout(cct, 5) << dendl;
 
-  if (!m_legacy_parent && r == -EOPNOTSUPP) {
+  if (!m_legacy_parent && r == -EOPNOTSUPP && !m_reattach) {
```

Comment:
I thought `set_parent` works for both v1 and v2 formats, and "legacy" case here is for handling older osds where `set_parent` method does not exists. No?

And if you worry about the case when need to reattach on older osds, then I don't think we need this, because currently the reattach is needed only for migration and it will not work with older osds anyway because of missing `set_migration` cls method.

Reply:
Indeed -- never mind. I forgot ```parent_attach``` was really just for namespace support, not clone v2.